### DISCUSSION TITLE
Run code after commit

### DISCUSSION THREAD

**Discussion by mwisnicki** [slick/slick#3006] [Questions]
Is it possible to attach callback to DBIO so that it will execute after commit?

**Comment by nafg** [comment]
Meaning to set a global hook that Slick will always use? No I don't think so.

What are you trying to accomplish?

**Reply by mwisnicki** [reply]
Not global. Specific to a single DBIO run. I want to update some metrics and write some logs but only if transaction is successful but it's pretty far from where the DBIO is actually run. 

I supposed I could carry it in `WriterT[DBIO, () => Unit, T]` but it would be a somewhat annoying so hoping there's something built-in to slick I can leverage.

**Reply by nafg** [reply]
Not sure I follow. I mean you can compose DBIOs obviously, so I don't think that's your question.

Can you sketch a program to illustrate what you want to accomplish?
